tfx/components/transform/component.py

```diff
@@ -103,6 +105,10 @@ def preprocessing_fn(inputs: Dict[Text, Any]) -> Dict[Text, Any]:
       transformed_examples: Optional output 'ExamplesPath' channel for
         materialized transformed examples, which includes both 'train' and
         'eval' splits.
+      cache_input_path: Optional input 'TransformCache' channel containing
+        cached information from previous Transform runs.
```

Comment:
could you check the cache folder structure? I feel it might need some change

e.g., 

run 1: tft takes span1, span2, span3, and generate cache for 123
run 2: tft takes span2, span3, span4, and generate cache for 234

for run2, it take the cache from resovler, which contains 1,2,3 (only 2 and 3 is useful, I wonder if it is already that smart to record the span information in cache outputs)

and the worse case, resolver of cache is not align with resolver of examples, e.g., example resolver returns 3 4 5, cache resolver returns 6 7 8, will it crash?

Reply:
before cache information is store with output artifact, so it might be 1:1 mapping.

the cache channel might also has multiple artifacts

Reply:
Resolved over chat conversation (i think), as Cache artifact stores whether a pattern has already been processed, not how a span has been prcession. Will resolve this once we hear back from Zohar.

Reply:
Zohar confirmed that Cache artifact is a DatasetKey (file pattern) to cache info structure (i.e. one output artifact). Resolved.